File: src/main/scala/org/xbery/artbeams/common/access/domain/EntityKey.scala
package org.xbery.artbeams.common.access.domain

import org.xbery.artbeams.common.assets.domain.AssetAttributes

/**
  * Unique key of an entity in the whole system.
 *
  * @author Radek Beran
  */
@SerialVersionUID(-1L)
case class EntityKey(
  /* Type of entity (system description). */
  entityType: String,
  /* Identification of an entity that is unique within given entity type. */
  entityId: String
) extends Serializable

object EntityKey {
  lazy val Empty = EntityKey(
    "",
    AssetAttributes.EmptyId
  )

  def fromClassAndId(cls: Class[_], id: String): EntityKey = EntityKey(cls.getSimpleName(), id)
}

File: src/main/kotlin/org/xbery/artbeams/common/access/domain/EntityKey.kt
package org.xbery.artbeams.common.access.domain

import org.xbery.artbeams.common.assets.domain.AssetAttributes
import java.io.Serializable

/**
 * Unique key of an entity in the whole system.
 *
 * @author Radek Beran
 */
data class EntityKey(val entityType: String, val entityId: String) : Serializable {
    companion object {
        val Empty: EntityKey = EntityKey("", AssetAttributes.EMPTY_ID)
        fun fromClassAndId(cls: Class<*>, id: String): EntityKey = EntityKey(cls.simpleName, id)
    }
}
